src/Features/header/components/CityLanguageSelect/City/CitySelect.tsx

```diff
@@ -0,0 +1,81 @@
+import Link from 'next/link';
+import React, { ReactElement, useState, useEffect } from 'react';
+import { setCookie, getCookie, CookieValueTypes } from 'cookies-next';
+import { Modal } from '../../../../../components/Modal/Modal';
+
+interface CitySelectProps {
+  myValue: string | undefined;
+}
+
+export const CitySelect: React.FC<CitySelectProps> = ({ myValue }): ReactElement => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [selectedValue, setSelectedValue] = useState<string | undefined>(myValue);
+
+  useEffect(() => {
+    const cookieValue = getCookie('myValue') as CookieValueTypes;
+    if (typeof cookieValue === 'string') {
+      setSelectedValue(cookieValue || undefined);
```

Comment:
I think after modifications above, you can delete `|| undefined`, see this is why it was not extra types above for `selectedValue`